cmd/ocm/create/user/cmd.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/spf13/cobra"
+
+	c "github.com/openshift-online/ocm-cli/pkg/cluster"
+)
+
+var args struct {
+	clusterKey string
+	group      string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "user",
+	Aliases: []string{"users"},
+	Short:   "Configure user access for cluster",
+	Long:    "Configure user access for cluster",
+	Example: `# Add a user to the dedicated-admins group
+  ocm create user myuser --cluster=mycluster --group=dedicated-admins`,
+	RunE: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVarP(
+		&args.clusterKey,
+		"cluster",
+		"c",
+		"",
+		"Name or ID of the cluster to add the user to (required).",
+	)
+	Cmd.MarkFlagRequired("cluster")
+
+	flags.StringVar(
+		&args.group,
+		"group",
+		"",
+		"Group name to add the users to.",
+	)
+	Cmd.MarkFlagRequired("group")
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	if !c.IsValidClusterKey(clusterKey) {
+		return fmt.Errorf(
+			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+	}
+
+	if len(argv) != 1 || argv[0] == "" {
+		return fmt.Errorf("At least one user must be specified")
+	}
+	users := argv[0]
+	// Load the configuration file:
+	cfg, err := config.Load()
+	if err != nil {
+		return fmt.Errorf("Can't load config file: %v", err)
+	}
+	if cfg == nil {
+		return fmt.Errorf("Not logged in, run the 'login' command")
+	}
+
+	// Check that the configuration has credentials or tokens that haven't have expired:
+	armed, err := cfg.Armed()
+	if err != nil {
+		return fmt.Errorf("Can't check if tokens have expired: %v", err)
+	}
+	if !armed {
+		return fmt.Errorf("Tokens have expired, run the 'login' command")
+	}
+
+	// Create the connection, and remember to close it:
+	connection, err := cfg.Connection()
+	if err != nil {
+		return fmt.Errorf("Can't create connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the cluster management api
+	clusterCollection := connection.ClustersMgmt().V1().Clusters()
+
+	cluster, err := c.GetCluster(clusterCollection, clusterKey)
+	if err != nil {
+		return fmt.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+	}
+
+	if cluster.State() != cmv1.ClusterStateReady {
+		return fmt.Errorf("Cluster '%s' is not yet ready", clusterKey)
+	}
+
+	_, err = clusterCollection.Cluster(cluster.ID()).
+		Groups().
+		Group(args.group).
+		Get().
+		Send()
+	if err != nil {
+		return fmt.Errorf("Group '%s' in cluster '%s' doesn't exist", args.group, clusterKey)
+	}
+
+	for _, username := range strings.Split(users, ",") {
+		user, err := cmv1.NewUser().ID(username).Build()
+		if err != nil {
+			return fmt.Errorf("Failed to create '%s' user '%s' for cluster '%s'", args.group, username, clusterKey)
```

Comment:
Does this work `Best Effort`?
We create users until we reach a user that we failed.
On first failure, we stop, and all the users that we succeeded are created.

say I had a list of 3 users and the second one already exists.
So the client will see:
```
> ocm create user u1,u2,u3 --cluster=mycluster --group=dedicated-admins`
{
  "kind": "User",
  "href": "/api/clusters_mgmt/v1/clusters/11111/groups/dedicated-admins/users/u1",
  "id": "u1"
}

Failed to add 'dedicated-admins' user 'u2' to cluster 'mycluster': user 'u2' already exists
```

I wonder if continuing would be better user experience. i.e:
```
> ocm create user u1,u2,u3 --cluster=mycluster --group=dedicated-admins`
{
  "kind": "User",
  "href": "/api/clusters_mgmt/v1/clusters/11111/groups/dedicated-admins/users/u1",
  "id": "u1"
}

Failed to add 'dedicated-admins' user 'u2' to cluster 'mycluster': user 'u2' already exists

{
  "kind": "User",
  "href": "/api/clusters_mgmt/v1/clusters/11111/groups/dedicated-admins/users/u3",
  "id": "u3"
}

```

WDYT?

Reply:
makes sense, I will change it

Reply:
This is marked as resolved but I don't see any change (or any commits after this message)